app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -116,6 +116,22 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       }
     },
 
+    populateLinkDomain: function (clickData) {
```

Comment:
I'm a bit uncertain here - this function doesn't seem to return anything? It might be working in some clever way that I've missed but generally for clarity I'd either have a function return something or have it act on a `this` variable.

Reply:
@andysellick objects are passed by reference, so the object passed through to the function gets modified directly. So `clickData` gets modified without the need to return anything. Should I change it so that it returns the updated values instead?

Reply:
I guess if it works okay and there's no risk of a problem then it's okay - I'm just not sure what the accepted practice is for this kind of situation.